### PR TITLE
fix: intermitent failure of creating data source of QuickSight

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [x] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/reporting/private/dashboard.ts [hunk 1]
```diff
@@ -34,6 +34,19 @@ export interface QuicksightCustomResourceProps {
   readonly redshiftProps: RedShiftProps;
 };
 
+export interface NetworkInterfaceCheckCustomResourceProps {
+  readonly networkInterfaces: string;
+  readonly vpcId: string;
+};
+
+export type NetworkInterfaceCheckCustomResourceLambdaProps = {
+  readonly awsAccountId: string;
```
Comment (zxkane) [inline src/reporting/private/dashboard.ts:43]: Only `networkInterfaces` is sufficient for querying the status of network interfaces.

File: src/reporting/private/dashboard.ts [hunk 2]
```diff
@@ -34,6 +34,19 @@ export interface QuicksightCustomResourceProps {
   readonly redshiftProps: RedShiftProps;
 };
 
+export interface NetworkInterfaceCheckCustomResourceProps {
+  readonly networkInterfaces: string;
+  readonly vpcId: string;
```
Comment (zxkane) [inline src/reporting/private/dashboard.ts:39]: vpcId is useless.

File: src/reporting/lambda/custom-resource/quicksight/network-interface-check.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    region,
+    ...aws_sdk_client_common_config,
+  });
+
+  if (event.RequestType === 'Create' || event.RequestType === 'Update' ) {
+    return _onCreate(ec2Client, props);
+  }
+};
+
+const _onCreate = async (ec2Client: EC2, props: NetworkInterfaceCheckCustomResourceLambdaPropsType): Promise<CdkCustomResourceResponse> => {
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/quicksight/network-interface-check.ts:43]: add unit test for this Lambda function

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/reporting/lambda/custom-resource/quicksight/network-interface-check.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  const networkInterfaceIds: string[] = [];
+  for (const ni of props.networkInterfaces) {
+    networkInterfaceIds.push(ni.NetworkInterfaceId);
+  }
+
+  logger.info(`networkInterfaceIds: ${networkInterfaceIds}`);
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/quicksight/network-interface-check.ts:50]: change to `logger.info('networkInterfaceIds: ', {networkInterfaceIds});`

File: test/reporting/quicksight/lambda/network-interface-check.test.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        {
+          NetworkInterfaceId: 'eni-test11113',
+          Status: NetworkInterfaceStatus.in_use,
+        },
+      ],
+    }).resolvesOnce({
+      NetworkInterfaces: [
+        {
+          NetworkInterfaceId: 'eni-test11111',
+          Status: NetworkInterfaceStatus.available,
```
Comment (zxkane) [inline test/reporting/quicksight/lambda/network-interface-check.test.ts:137]: Why do we only consider the `in_use` not `available` ENI ready for creating data sources in QuickSight?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: test/reporting/quicksight/lambda/network-interface-check.test.ts (hunk 5 above)
Comment (llmin) [reply test/reporting/quicksight/lambda/network-interface-check.test.ts:137]: If the network interface is not attached to an instance, the status is available ; 
if a network interface is attached to an instance the status is in-use .

We  consider in-use to make sure that QuickSight VPC connection is 100% percent ready to use.

**Review by zxkane — APPROVED** [review]

**Pull request merged**